src/page/profile.rs

```diff
@@ -1,35 +1,37 @@
 use super::ViewPage;
-use crate::entity::{
-    article::{self, Article},
-    author::{self, Author},
-    ErrorMessage, PageNumber, PaginatedList, Username, Viewer,
-};
+use std::borrow::Cow;
+
+use seed::prelude::*;
+
 use crate::{
+    entity::{
+        article::{self, Article},
+        author::{self, Author},
+        ErrorMessage, PageNumber, PaginatedList, Username, Viewer,
+    },
     helper::take,
     loading, logger, page, request,
     route::{self, Route},
     GMsg, Session,
 };
-use futures::prelude::*;
-use seed::prelude::*;
-use std::borrow::Cow;
 
 static DEFAULT_TITLE_PREFIX: &str = "Profile";
 static TITLE_PREFIX_FOR_ME: &str = "My Profile";
 
-fn fetch_feed(
+async fn fetch_feed(
     viewer: Option<Viewer>,
     username: Username<'static>,
     selected_feed: SelectedFeed,
     page_number: PageNumber,
-) -> impl Future<Item = Msg, Error = Msg> {
+) -> Result<Msg, Msg> {
```

Comment:
Question: There are many `Result<Msg, Msg>` / `Result<Ms, Ms>` - do you think we can merge it to something like `ResultMsg<Ms>` (or is there an idiomatic way how to express that success and fail have the same type)? Does it make sense to you?

Reply:
I guess the only thing to do is a type alias, but I don't think that adds any value.